### PR TITLE
prov/gni: implement MSG EP CM data

### PR DESCRIPTION
**Pull request by ztiffany**
Signed-off-by: Zach <ztiffany@cray.com>

@sungeunchoi @hppritcha

### CODE REVIEW

File: prov/gni/src/gnix_cm.c [hunk 1]
```diff
@@ -360,6 +363,12 @@ DIRECT_FN STATIC int gnix_connect(struct fid_ep *ep, const void *addr,
 	req.cq_irq_mdh = ep_priv->nic->irq_mem_hndl;
 	req.peer_caps = ep_priv->caps;
 
+	req.cm_data_len = paramlen;
+	if (paramlen) {
```
Comment (hppritcha) [inline prov/gni/src/gnix_cm.c:378]: should there be a check for paramlen being too large?  or is this checked somewhere else?

File: prov/gni/src/gnix_cm.c [hunk 2]
```diff
@@ -471,6 +480,12 @@ DIRECT_FN STATIC int gnix_accept(struct fid_ep *ep, const void *param,
 	resp.cq_irq_mdh = ep_priv->nic->irq_mem_hndl;
 	resp.peer_caps = ep_priv->caps;
 
+	resp.cm_data_len = paramlen;
+	if (paramlen) {
```
Comment (hppritcha) [inline prov/gni/src/gnix_cm.c:497]: same check for paramlen here too?

File: prov/gni/test/cm.c [hunk 3]
```diff
@@ -68,13 +68,18 @@
...
+
+
 static struct fid_fabric *cli_fab;
 static struct fid_domain *cli_dom;
 static struct fid_ep *cli_ep;
 static struct fi_info *cli_hints;
 static struct fi_info *cli_fi;
 static struct fid_eq *cli_eq;
 static struct fid_cq *cli_cq;
+char *cli_cm_in_data = "Hola.  Soy cliente.";
```
Comment (hppritcha) [inline prov/gni/test/cm.c:82]: cute

**Comment by hppritcha** [comment]
except for questions about paramlen checking looks good.

**Comment by ztiffany** [comment]
@hppritcha I added paramlen checks.

File: prov/gni/test/cm.c (hunk 3 above)
Comment (sungeunchoi) [reply prov/gni/test/cm.c:82]: @ztiffany are you planning another vacation to Central America?

File: prov/gni/test/cm.c [hunk 4]
```diff
@@ -68,13 +69,18 @@
 
 #define DEF_PORT "1973"
 
+#define CM_DATA_SIZE	256
```
Comment (sungeunchoi) [inline prov/gni/test/cm.c]: Does this have to be the same as GNIX_CM_DATA_MAX_SIZE?  If so, why not use GNIX_CM_DATA_MAX_SIZE directly?

**Comment by sungeunchoi** [comment]
Does this get any new fabtests passing?

**Comment by hppritcha** [comment]
bot:lanl:retest

**Comment by hppritcha** [comment]
bot:lanl:retest

File: prov/gni/test/cm.c (hunk 4 above)
Comment (ztiffany) [reply prov/gni/test/cm.c]: I can change this.

**Comment by ztiffany** [comment]
@sungeunchoi I don't think any new fabtests pass, but some are closer to passing.

**Comment by hppritcha** [comment]
:+1:
i'd suggest merge after travis CI done.

**Comment by hppritcha** [comment]
apparently the content of this PR already got merged upstream somehow.

**Pull request merged**